### PR TITLE
Doc improvement Enum

### PR DESCRIPTION
**Pull request by jonathanrocher**
1. Fix syntax highlighting in existing code snippet for `Enum`
2. Added a note about making the list of values dynamic

### CODE REVIEW

File: docs/source/traits_user_manual/defining.rst [hunk 1]
```diff
@@ -480,8 +480,7 @@ value.
 
 A trait defined in this fashion can accept only values that are contained in
 the list of permitted values. The default value is the first value specified;
-it is also a valid value for assignment.
-::
```
Comment (jonathanrocher) [inline docs/source/traits_user_manual/defining.rst]: This was preventing syntax highlighting.

File: docs/source/traits_user_manual/defining.rst [hunk 2]
```diff
@@ -500,20 +499,46 @@ it is also a valid value for assignment.
     >>> hats.stock = 'many' # OK
     >>> hats.stock = 4      # Error, value is not in \
     >>>                     # permitted list
-    Traceback (most recent call last):
-      File "<stdin>", line 1, in <module>
-      File "c:\svn\ets3\traits_3.0.3\enthought\traits\trait_handlers.py", line 175,
```
Comment (jonathanrocher) [inline docs/source/traits_user_manual/defining.rst]: This was showing how old the doc was :)

File: docs/source/traits_user_manual/defining.rst [hunk 3]
```diff
@@ -500,20 +499,46 @@ it is also a valid value for assignment.
...
 
-This example defines an InventoryItem class, with two trait attributes,
+
+This defines an :py:class:`InventoryItem` class, with two trait attributes,
 **name**, and **stock**. The name attribute is simply a string. The **stock**
 attribute has an initial value of None, and can be assigned the values None, 0,
 1, 2, 3, and 'many'. The example then creates an instance of the InventoryItem
 class named **hats**, and assigns values to its attributes.
 
+When the list of possible values can change during the lifetime of the object,
```
Comment (jonathanrocher) [inline docs/source/traits_user_manual/defining.rst]: This is the new stuff.

File: docs/source/traits_user_manual/defining.rst (hunk 1 above)
Comment (mdickinson) [inline docs/source/traits_user_manual/defining.rst]: Hmm.  That's valid reST, and shouldn't prevent the syntax highlighting.  And indeed, if I restore the construct in this PR, the syntax highlighting remains.  I suspect that it's the changes below that actually make the difference to the highlighting.

File: docs/source/traits_user_manual/defining.rst [hunk 4]
```diff
@@ -500,20 +499,46 @@ it is also a valid value for assignment.
...
+one can specifies **another trait** that holds the list of possible values as
```
Comment (mdickinson) [inline docs/source/traits_user_manual/defining.rst]: `specify`

File: docs/source/traits_user_manual/defining.rst [hunk 5]
```diff
@@ -500,20 +499,46 @@ it is also a valid value for assignment.
...
+so::
+
+    >>> from traits.api import Enum, HasTraits, List
+    >>> class InventoryItem(HasTraits):
+    ...    possible_stock_states  = List([None, 0, 1, 2, 3, 'many'])
+    ...    stock = Enum(None, values="possible_stock_states")
+    ...            # Enumerated list, default value is 'None'. The list of
+    ...            # allowed values is whatever possible_stock_states holds
```
Comment (mdickinson) [inline docs/source/traits_user_manual/defining.rst]: How about using a different default here (`0`, say), to make it clear that the rule above that "The first item in the list is used as the default" doesn't apply when you're specifying a default directly?

**Comment by mdickinson** [comment]
LGTM.  I made a few tweaks in 4b48c29264b436b0c2692d1d7bdb94f5945322d0; let me know what you think.

**Comment by mdickinson** [comment]
BTW, I suspect that the missing syntax highlighting was a result of Pygments failing to parse the code snippet (possibly because of the contents of the traceback, but who knows).  There are other snippets in this document that are also missing the highlighting.

**Comment by mdickinson** [comment]
Syntax highlighting now fixed for all snippets in this file.

**Comment by jonathanrocher** [comment]
LGTM. Thanks for the improvements...

**Comment by mdickinson** [comment]
Merging.  Thanks!

**Pull request merged**